### PR TITLE
nixos/networking-interfaces: change preferTempAddress to allow disabling temp addresses

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Fix issue #75311

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested `nixosTests.networking.scripted.privacy` `nixosTests.networking.networkd.privacy`
- [x] Built manual
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
@GrahamcOfBorg test networking.scripted.privacy networking.networkd.privacy

File: nixos/modules/tasks/network-interfaces.nix [hunk 1]
```diff
@@ -973,8 +999,12 @@ in
       "net.ipv6.conf.all.forwarding" = mkDefault (any (i: i.proxyARP) interfaces);
     } // listToAttrs (flip concatMap (filter (i: i.proxyARP) interfaces)
         (i: forEach [ "4" "6" ] (v: nameValuePair "net.ipv${v}.conf.${replaceChars ["."] ["/"] i.name}.proxy_arp" true)))
-      // listToAttrs (forEach (filter (i: i.preferTempAddress) interfaces)
-        (i: nameValuePair "net.ipv6.conf.${replaceChars ["."] ["/"] i.name}.use_tempaddr" 2));
+      // listToAttrs (forEach interfaces
+        (i: let
+          opt = i.tempAddress;
+          val = if opt == "disabled" then 0 else
```
Comment (lheckemann) [inline nixos/modules/tasks/network-interfaces.nix]: `{disabled = 0; enabled = 1; default = 2;}.${opt}` might be clearer?

**Review by lheckemann — APPROVED** [review]
Looks well documented, well thought out and well implemented overall (just one minor nitpick, see inline). I haven't tested this, but if the tests pass I'm happy to merge this with my suggestion applied or as is. Thanks!

**Comment by lheckemann** [comment]
@GrahamcOfBorg test networking.scripted.privacy networking.networkd.privacy

File: nixos/modules/tasks/network-interfaces.nix (hunk 1 above)
Comment (rnhmjoj) [reply nixos/modules/tasks/network-interfaces.nix]: Oh, nice! I didn't think about this.

**Comment by rnhmjoj** [comment]
Ok, I applied your suggestion. The tests passes on my end. Let's test again with:

@GrahamcOfBorg test networking.scripted.privacy networking.networkd.privacy

**Comment by rnhmjoj** [comment]
Thank you!

**Comment by volth** [comment]
tinc interfaces still have privacy ipv6 addresses (like in https://github.com/NixOS/nixpkgs/issues/75311#issuecomment-563297293) even when 
`networking.interfaces."tinc.dmz".tempAddress = "disable"`
:(

**Comment by volth** [comment]
`"net.ipv6.conf.${interface}.addr_gen_mode" = 1` does remove that "stable-privacy" address

interesting that only `tinc` gets the "stable-privacy" address, other vpn interfaces (`wireguard` for example) do not

**Pull request merged**